src/modules/lists/lists.controller.ts

```diff
@@ -70,4 +91,34 @@ export class ListsController {
     };
   }
 
+  @ApiOperation({ title: "Delete the given mentor's list for the given user" })
+  @Delete(':listId')
+  async deleteList(
+    @Req() request: Request,
+    @Param('userid') userId: string,
+    @Param('listId') listId: string,
+  ) {
+    const current: User = await this.usersService.findByAuth0Id(
+      request.user.auth0Id,
+    );
+    const user: User = await this.usersService.findById(userId);
+
+    if (!current._id.equals(user._id) && !current.roles.includes(Role.ADMIN)) {
```

Comment:
please add an addition check if the given `user` exist, else throw a `BadRequestException` exception before getting to this line

Reply:
Done